File: Algorithms/python/001 - Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashmap = {}
        temp = []
        for i,x in enumerate(nums):
            if x in hashmap:
                temp.append(hashmap[x])
                temp.append(i)
                break
            hashmap[target - x] = i
        
        return temp

File: Algorithms/cpp/001 - Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        vector<int> result;
        for (int i = 0; i < nums.size(); i++) {
            if (m.find(nums[i]) != m.end()) {
        		result.push_back(m[nums[i]]);
        		result.push_back(i);
        		break;
        	}
        	m[target - nums[i]] = i;
        }
        return result;
    }
};